### PR TITLE
Update library for new systemd install

### PR DESCRIPTION
**Pull request by sagesyr**
- Updates the [Run Teleport as a Daemon](https://goteleport.com/docs/setup/admin/daemon/) guide to reflect changes to the process of setting up a daemon when using the new `telelport install systemd` command
- Adds new daemon install steps to our [Enterprise Getting Started Guide](https://goteleport.com/docs/enterprise/getting-started/)
- Adds new daemon install steps  to our [Getting Started on a Linux Server Guide](https://goteleport.com/docs/getting-started/linux-server/)

Followup to https://github.com/gravitational/teleport/pull/15602, resolves commit history issue related to merge conflict

### CODE REVIEW

File: docs/pages/deploy-a-cluster/teleport-enterprise/getting-started.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Teleport Enterprise Quick Start
 description: How to set up and configure Teleport Enterprise for SSH
-h1: Teleport Enterprise Quick Start
+# Teleport Enterprise Quick Start
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/teleport-enterprise/getting-started.mdx:4]: Is this necessary? It's the same as the title, so can we just remove it?

File: docs/pages/management/admin/daemon.mdx [hunk 2]
```diff
@@ -2,18 +2,16 @@
 title: Run Teleport as a Daemon
 description: Configure Teleport to run as a daemon using systemd
 ---
+On Linux systems in non-containerized environments, we recommend running the `teleport` binary as a daemon using
+systemd. Using a daemon helps to ensure that the `teleport` process can remain running and available regardless of a controlling terminal session or parent process, and opens up additional
```
Comment (zmb3) [inline docs/pages/management/admin/daemon.mdx:6]: Consider wrapping these long lines. `reflow-markdown` is a nice VS Code plugin that will do it for you.

File: docs/pages/management/admin/daemon.mdx [hunk 3]
```diff
@@ -33,89 +31,101 @@ $ readlink /sbin/init
...
 
-</Notice>
+  ```code
+  ● teleport.service - Teleport SSH Service
+       Loaded: loaded (/lib/systemd/system/teleport.service; disabled; vendor preset: enabled)
+       Active: inactive (dead)
+  ```
 
-Enable the unit so systemd can place it in its dependency tree:
+  Enter the following command to enable the systemd `teleport` daemon. Enabling the daemon using `systemctl` allows systemd to place the daemon in its dependency tree, ensuring that the teleport service will be automatically restarted on any potential reboot:
```
Comment (zmb3) [inline docs/pages/management/admin/daemon.mdx:74]:
```suggestion
  Enter the following command to enable the systemd `teleport` daemon. Enabling the daemon using `systemctl` allows systemd to place the daemon in its dependency tree, ensuring that the Teleport service will be automatically restarted on any potential reboot:
```

If you're referring to the product in general and not specifically to the `teleport` program, I think this should be capitalized, right?

File: docs/pages/management/admin/daemon.mdx [hunk 4]
```diff
@@ -33,89 +31,101 @@ $ readlink /sbin/init
...
+  Active: active (running) since Mon 2022-04-18 18:33:41 UTC; 41s ago
+  ```
+  </TabItem>
+
+  <TabItem label="Source Installation">
+
+  Enter the following `teleport` command to have Teleport output a systemd unit file to `stdout` to be reviewed and applied as needed:
+
+  ```code
+  $ sudo teleport install systemd
```
Comment (zmb3) [inline docs/pages/management/admin/daemon.mdx:99]: No need for `sudo` here. That's kind of the point of allowing users to view the configuration first before they run a privileged command to write it to a system location.

File: docs/pages/management/admin/daemon.mdx [hunk 5]
```diff
@@ -143,7 +153,33 @@ To upgrade a host to a newer version of Teleport, you must:
...
 
-## Further reading
+
+## Understanding Teleport daemon command line options for unit file configuration
+
+The `teleport install systemd` command includes a number of optional flags that you can use to strictly define parameters of the generated unit file and set where that output will be written.
+The following table includes all command line options available with the `teleport install systemd` command, a brief description of what they do, and our general recommended defaults tested on **Ubuntu 20.04 LTS**:
+
+
+| Flag | Description | General Recommended Default |
```
Comment (zmb3) [inline docs/pages/management/admin/daemon.mdx:163]: Hmm, "General Recommended Default" sounds different than "Default". It makes it seem like the user should set these defaults if they don't know what else to put, when in reality Teleport will just do it for them.

Maybe change the column to "Default" instead?

File: docs/pages/management/admin/daemon.mdx [hunk 6]
```diff
@@ -143,7 +153,33 @@ To upgrade a host to a newer version of Teleport, you must:
...
+| --- | ----------- | ---- |
+| `--fd-limit` | The maximum number of open file descriptors by the service, defined by `LimitNOFILE` in the unit file. |  `--fd-limit=8192` |
+| `--env-file` | The path to the `EnvironmentFile` defined by the system unit file, containing any potential variables to be used in the configuration. | `--envfile=/etc/default/teleport` |
+| `--pid-file` | The path to the pid file containing the process identification number. | `--pid-file=/run/teleport.pid` |
+| `--teleport-path` | The full path to the `teleport` binary. | `--teleport-path=/usr/local/bin/teleport` |
+| `-o`, `--output` | The full unit file load path which will define where to create and store the system unit file itself. This flag is used to write the output to the designated file. | `--output=/etc/systemd/system/teleport.service` |
+
+If using the recommended defaults, you can enter all command line options in a single command as follows:
```
Comment (zmb3) [inline docs/pages/management/admin/daemon.mdx:171]: If using the recommended defaults, then you don't need to enter all these flags manually, a simple `sudo teleport install systemd -o /etc/systemd/system/teleport.service` will do the trick.

**Review by zmb3 — COMMENTED** [review]
Looking good, just a few minor comments.

File: docs/pages/management/admin/daemon.mdx (hunk 6 above)
Comment (rsyracuse) [reply docs/pages/management/admin/daemon.mdx]: Mentioned this but kept the example the same, let me know if you'd like to adjust it differently

File: docs/pages/management/admin/daemon.mdx (hunk 4 above)
Comment (rsyracuse) [reply docs/pages/management/admin/daemon.mdx]: That makes sense, happy to take care of that

File: docs/pages/management/admin/daemon.mdx (hunk 2 above)
Comment (rsyracuse) [reply docs/pages/management/admin/daemon.mdx]: Great tool! Will be using that from now on

File: docs/pages/deploy-a-cluster/teleport-enterprise/getting-started.mdx (hunk 1 above)
Comment (rsyracuse) [reply docs/pages/deploy-a-cluster/teleport-enterprise/getting-started.mdx]: Not to my knowledge, and I can see others removing it entirely within our repo. Removing as well

**Review by klizhentas — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

**Comment by ptgott** [comment]
@rsyracuse This is good to merge whenever you're ready! This will also create a backport PR for v10—we can indicate in that backport if this feature is only available for a certain minor version of v10.

**Pull request merged**